src/TensorFlowNET.Keras/Engine/Model.Evaluate.cs

```diff
@@ -46,18 +55,31 @@ public partial class Model
                 StepsPerExecution = _steps_per_execution
             });
 
+            var callbacks = new CallbackList(new CallbackParams
+            {
+                Model = this,
+                Verbose = verbose,
+                Steps = data_handler.Inferredsteps
+            });
+            callbacks.on_test_begin();
+
             foreach (var (epoch, iterator) in data_handler.enumerate_epochs())
             {
                 reset_metrics();
-                // callbacks.on_epoch_begin(epoch)
+                //callbacks.on_epoch_begin(epoch);
                 // data_handler.catch_stop_iteration();
-                IEnumerable<(string, Tensor)> results = null;
+                IEnumerable<(string, Tensor)> logs = null;
+
                 foreach (var step in data_handler.steps())
                 {
-                    // callbacks.on_train_batch_begin(step)
-                    results = test_function(data_handler, iterator);
+                    callbacks.on_train_batch_begin(step);
```

Comment:
Should this be `on_test_batch_begin`?

Reply:
> Should this be `on_test_batch_begin`?

sry, there are some mistakes happened, I will correct it.